### PR TITLE
Add some default values

### PR DESCRIPTION
**Pull request by ashish-amarnath**
Signed-off-by: Ashish Amarnath <ashish.amarnath@gmail.com>

### CODE REVIEW

File: cmd/testdata/default-capa-no-machine-deployment.golden [hunk 1]
```diff
@@ -80,7 +88,10 @@ spec:
     localAPIEndpoint:
       advertiseAddress: ""
       bindPort: 0
-    nodeRegistration: {}
+    nodeRegistration:
+      kubeletExtraArgs:
+        cloud-provider: aws
+      name: '''{{ ds.meta_data.hostname }}'''
```
Comment (chuckha) [inline cmd/testdata/default-capa-no-machine-deployment.golden]: This looks a little funny, I'd expect to see `'{{ ds.meta_data.hostname }}'`

Comment (ashish-amarnath) [reply cmd/testdata/default-capa-no-machine-deployment.golden]: It does look out of place! It is templating for the cloud-init configuration.

Comment (chuckha) [reply cmd/testdata/default-capa-no-machine-deployment.golden]: i meant the `'''` part

**Comment by ashish-amarnath** [comment]
I think the messiness is due to the, IMO unavoidable, coupling between the infra provider and the bootstrap provider.

I chose to pass the `ip` to the bootstrap provider interface because I didn't want to make the  "default setting" as an optional step. 

I think we should avoid `switch infraProvider {` to be sprinkled in multiple places. i.e. once we know what infra provider is being used, all the decisions taking infra provider into account can be made. Also, it may be best to encapsulate all infra provider-specific methods into one interface instead of creating multiple interfaces.

That said, we can rename `UpdateBootstrapConfig` to `SetBootstrapConfigInfraValues()` and call it on the infraProvider if `--skip-defaults` is not used. This will encapsulate all infra provider related methods into one interface.

WDYT?

**Comment by chuckha** [comment]
Agreed that there's no logical reason to switch on infraProvider more than once.

I think there's no reason to even allow for `--skip-defaults` if we embed the defaulter directly into the infrastructure provider. It doesn't even have to be a function call, just default values we generate as part of the config. If we find a use case for *not* providing defaults we can add a `--skip-defaults` flag later.

Honestly I always want a working config for a given infrastructure provider and never a "blank" template, that's just not useful.

The outlined approach sounds fine to me, ping me know when you'd like another review!

**Comment by chuckha** [comment]
I think we should merge this and work on the abstraction later. I really need defaults in my YAML haha

**Comment by ashish-amarnath** [comment]
Yeah.. I’ll finish this up today.

**Pull request merged**